### Partie 1 – Explorer les données 

##### 1) Charger les données CSV 

In [2]:
import numpy as np
import pandas as pd 

In [3]:
df = pd.read_csv("../data/smart_building_raw.csv")
print(df)

     id_mesure                 date batiment      type_batiment zone  \
0         1174  2025-02-13 06:00:00       B8           Entrepôt    A   
1         1275  2025-03-10 12:00:00       B7             Bureau    D   
2         1493  2025-05-04 00:00:00       B5  Centre commercial    B   
3         1073  2025-01-19 00:00:00       B6         Université    D   
4         1454  2025-04-24 06:00:00       B7             Bureau    D   
..         ...                  ...      ...                ...  ...   
502       1107  2025-01-27 12:00:00       B2              École    C   
503       1271  2025-03-09 12:00:00       B3            Hôpital    B   
504       1349  2025-03-29 00:00:00       B5  Centre commercial    B   
505       1436  2025-04-19 18:00:00       B7             Bureau    C   
506       1103  2025-01-26 12:00:00       B2              École    D   

     temperature  humidite     co2  occupation  consommation_kwh  \
0           23.0      40.0   851.0        26.0             138.4   

##### 2) Afficher les premières lignes du dataset

In [4]:
print(df.head())

   id_mesure                 date batiment      type_batiment zone  \
0       1174  2025-02-13 06:00:00       B8           Entrepôt    A   
1       1275  2025-03-10 12:00:00       B7             Bureau    D   
2       1493  2025-05-04 00:00:00       B5  Centre commercial    B   
3       1073  2025-01-19 00:00:00       B6         Université    D   
4       1454  2025-04-24 06:00:00       B7             Bureau    D   

   temperature  humidite     co2  occupation  consommation_kwh  \
0         23.0      40.0   851.0        26.0             138.4   
1         28.0      69.0   538.0         0.0             170.2   
2         19.5      52.3  1198.0        64.0             149.8   
3         20.0      58.4  1014.0        30.0             141.6   
4         26.8      70.5   628.0        42.0             228.3   

  mode_climatisation etat_systeme jour_semaine alerte  
0                Eco       Normal        Jeudi    Non  
1                Eco       Normal        Lundi    Non  
2             

##### 3) Afficher les dernières lignes du dataset 

In [5]:
print(df.tail())

     id_mesure                 date batiment      type_batiment zone  \
502       1107  2025-01-27 12:00:00       B2              École    C   
503       1271  2025-03-09 12:00:00       B3            Hôpital    B   
504       1349  2025-03-29 00:00:00       B5  Centre commercial    B   
505       1436  2025-04-19 18:00:00       B7             Bureau    C   
506       1103  2025-01-26 12:00:00       B2              École    D   

     temperature  humidite     co2  occupation  consommation_kwh  \
502         29.8      66.5   778.0        48.0             243.5   
503         21.1      66.3  3900.0        53.0             145.8   
504         21.6      61.9   959.0        33.0             100.2   
505         21.2      54.5  1066.0        38.0             150.9   
506          NaN      64.8   584.0        29.0             163.4   

    mode_climatisation etat_systeme jour_semaine alerte  
502             Normal       Normal        Lundi    Non  
503                Eco       Alerte     Di

##### 4) Combien d'observations contient le dataset 

In [6]:
df.shape[0]
print ("Le nombre d'observation que contient le dataset est : 507")

Le nombre d'observation que contient le dataset est : 507


##### 5) Combien de variables possède le dataset 

In [7]:
df.columns.tolist()

['id_mesure',
 'date',
 'batiment',
 'type_batiment',
 'zone',
 'temperature',
 'humidite',
 'co2',
 'occupation',
 'consommation_kwh',
 'mode_climatisation',
 'etat_systeme',
 'jour_semaine',
 'alerte']

##### 6) Identifier les variables numériques 

In [8]:
var_num = df.select_dtypes(include=np.number).columns.tolist()
print(var_num)

['id_mesure', 'temperature', 'humidite', 'co2', 'occupation', 'consommation_kwh']


##### 7) Identifier les variables catégorielles 

In [9]:
print("les variables categorielles sont :", df.select_dtypes(include="object").columns.tolist())

les variables categorielles sont : ['date', 'batiment', 'type_batiment', 'zone', 'mode_climatisation', 'etat_systeme', 'jour_semaine', 'alerte']


C:\Users\dell\AppData\Local\Temp\ipykernel_6248\3312097636.py:1: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  print("les variables categorielles sont :", df.select_dtypes(include="object").columns.tolist())


##### 8) Identifier les dates 

In [10]:
print("les variables de type date:\n", df['date'])

les variables de type date:
 0      2025-02-13 06:00:00
1      2025-03-10 12:00:00
2      2025-05-04 00:00:00
3      2025-01-19 00:00:00
4      2025-04-24 06:00:00
              ...         
502    2025-01-27 12:00:00
503    2025-03-09 12:00:00
504    2025-03-29 00:00:00
505    2025-04-19 18:00:00
506    2025-01-26 12:00:00
Name: date, Length: 507, dtype: str


##### 9) Identifier les identifiants

In [11]:
print("les identifiants sont:\n", df['id_mesure'])

les identifiants sont:
 0      1174
1      1275
2      1493
3      1073
4      1454
       ... 
502    1107
503    1271
504    1349
505    1436
506    1103
Name: id_mesure, Length: 507, dtype: int64


##### 10) Déterminer les statistiques : moyenne, médiane, minimum, maximum, écart-type et quartiles. 

In [12]:
print("les statistiques pour les variables numeriques sont : \n", df.describe())

les statistiques pour les variables numeriques sont : 
          id_mesure  temperature    humidite          co2  occupation  \
count   507.000000   495.000000  496.000000   500.000000  501.000000   
mean   1251.114398    24.154141   57.864113   844.150000   44.850299   
std     144.782769     7.418465   16.026336   582.181386   24.949139   
min    1001.000000   -30.000000  -12.000000    89.000000  -20.000000   
25%    1125.500000    21.600000   49.275000   623.750000   27.000000   
50%    1252.000000    24.000000   57.550000   787.500000   46.000000   
75%    1376.500000    26.000000   65.750000   952.000000   61.000000   
max    1500.000000    96.000000  160.000000  6000.000000  116.000000   

       consommation_kwh  
count        502.000000  
mean         169.069323  
std           53.164294  
min         -100.000000  
25%          136.875000  
50%          169.800000  
75%          202.975000  
max          336.200000  


In [13]:
print("les statistiques pour les variables categorielles sont :\n ", df.describe(include="object"))

les statistiques pour les variables categorielles sont :
                         date batiment type_batiment zone mode_climatisation  \
count                   507      507           503  507                502   
unique                  500        8            16    4                  7   
top     2025-01-30 06:00:00       B1        Bureau    B             Normal   
freq                      2       94           210  143                271   

       etat_systeme jour_semaine alerte  
count           507          502    507  
unique            3            7      2  
top          Normal     Vendredi    Non  
freq            442           75    361  


C:\Users\dell\AppData\Local\Temp\ipykernel_6248\3490907401.py:1: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  print("les statistiques pour les variables categorielles sont :\n ", df.describe(include="object"))


##### 11) Y a-t-il des variables potentiellement problématiques 

Oui il y'a plusieurs variables potentiellement problematiques .
--on observe qu'il y'a des valeurs manquantes dans toutes les variables.

id_mesure a un count de 507, alors que les autres colonnes ont entre 495 et 502 valeurs non nulles. Donc il manque environ 12 valeurs en température, 11 en humidité, 7 en CO2, 6 en occupation et 5 en consommation.

--On a probablement la presences de doublons sur id_mesure.Les identifiants vont de 1001 à 1500, soit 500 valeurs possibles, mais il y a 507 lignes. Il y a donc probablement des lignes en double.

--On constate egalement des valeurs aberrantes:

temperature : de -30 à 96 °C, alors que 50 % des valeurs sont entre 21,6 et 26. Ces extrêmes ressemblent à des erreurs de capteur.

humidite : de -12 à 160, alors qu'une humidité en % doit rester entre 0 et 100.

co2 : le max de 6000 est très loin du 75e percentile (952), et l'écart-type (582) est énorme par rapport à la médiane. Le min de 89 est aussi suspect, car l'air extérieur est déjà autour de 400 ppm.

occupation : le min de -20 est impossible. car on ne peut pas avoir un nombre negatif de personnes.

consommation_kwh : le min de -100 est impossible (une consommation ne peut pas être négative).

##### 12) Pour les données incohérentes a) rechercher des valeurs telles que humidité < 0 

In [14]:
print("les valeurs telles que l'humidite est inferieur a 0 sont:\n", df[df["humidite"] < 0])

les valeurs telles que l'humidite est inferieur a 0 sont:
      id_mesure                 date batiment type_batiment zone  temperature  \
281       1126  2025-02-01 06:00:00       B2         École    D         24.8   
335       1036  2025-01-09 18:00:00       B4        Bureau    D         21.2   
366       1216  2025-02-23 18:00:00       B3       Hôpital    B         23.2   

     humidite    co2  occupation  consommation_kwh mode_climatisation  \
281      -5.0  759.0        57.0             225.2              Boost   
335      -8.0  160.0        25.0             120.2             Normal   
366     -12.0  702.0        41.0             119.8             Normal   

    etat_systeme jour_semaine alerte  
281       Normal       Samedi    Non  
335       Alerte        Jeudi    Non  
366       Normal     Dimanche    Non  


#####  12) Pour les données incohérentes b) rechercher des valeurs telles que humidité > 100

In [15]:
print("les valeurs telles que l'humidite est superieur a 100 sont:\n", df[df["humidite"] > 100])

les valeurs telles que l'humidite est superieur a 100 sont:
      id_mesure                 date batiment      type_batiment zone  \
59        1246  2025-03-03 06:00:00       B5  Centre commercial    A   
103       1016  2025-01-04 18:00:00       B6         Université    B   
128       1156  2025-02-08 18:00:00       B1             Bureau    A   
160       1186  2025-02-16 06:00:00       B8           Entrepôt    B   
268       1276  2025-03-10 18:00:00       B1             Bureau    D   
327       1066  2025-01-17 06:00:00       B2              École    C   
342       1096  2025-01-24 18:00:00       B8           Entrepôt    A   

     temperature  humidite    co2  occupation  consommation_kwh  \
59          24.2     108.0  813.0        18.0             -15.0   
103         26.9     145.0  670.0        28.0             175.9   
128          NaN     160.0  941.0        37.0             149.8   
160         24.9     125.0    NaN        18.0             123.5   
268         26.0     140.0 

##### 12) c) rechercher des valeurs telles que température extrêmement élevée 

In [16]:
print("les valeurs telles que la temperature est extremement elevee:\n", df[df["temperature"] > 40])

les valeurs telles que la temperature est extremement elevee:
      id_mesure                 date batiment      type_batiment zone  \
7         1141  2025-02-05 00:00:00       B2              École    D   
116       1181  2025-02-15 00:00:00       B6         Université    A   
161       1061  2025-01-16 00:00:00       B5  Centre commercial    D   
499       1021  2025-01-06 00:00:00       B2              École    A   

     temperature  humidite    co2  occupation  consommation_kwh  \
7           72.5      78.3  257.0       100.0             239.7   
116         96.0      86.5  874.0       115.0             241.5   
161         88.0      50.5  393.0        62.0             196.6   
499         95.2      65.0  314.0        43.0             246.3   

    mode_climatisation etat_systeme jour_semaine alerte  
7                Boost       Alerte     Mercredi    Non  
116             Normal       Normal       Samedi    Non  
161             Normal       Alerte        Jeudi    Non  
499     

##### 12) d) rechercher des valeurs telles que occupation négative

In [17]:
print("les valeurs telles que occupation negative :\n", df[df["occupation"] < 0])

les valeurs telles que occupation negative :
      id_mesure                 date batiment      type_batiment zone  \
22        1031  2025-01-08 12:00:00       B2              École    D   
376       1231  2025-02-27 12:00:00       B7             Bureau    D   
430       1081  2025-01-21 00:00:00       B1             Bureau    B   
487       1131  2025-02-02 12:00:00       B1           entrepot    A   
494       1331  2025-03-24 12:00:00       B5  Centre commercial    B   

     temperature  humidite     co2  occupation  consommation_kwh  \
22          26.1      57.6   648.0        -5.0             170.0   
376         19.0      35.3   680.0        -8.0             167.9   
430         23.5      44.1   541.0       -12.0             248.7   
487         26.3      70.6  1519.0        -2.0             173.4   
494         22.7      54.6   979.0       -20.0             178.6   

    mode_climatisation etat_systeme jour_semaine alerte  
22                 Eco       Normal     Mercredi    No

##### 12) e) rechercher des valeurs telles que consommation négative

In [18]:
print("les valeurs telles que consommation negative :\n", df[df["consommation_kwh"] < 0])

les valeurs telles que consommation negative :
      id_mesure                 date batiment      type_batiment zone  \
59        1246  2025-03-03 06:00:00       B5  Centre commercial    A   
154       1046  2025-01-12 06:00:00       B2              École    B   
199       1146  2025-02-06 06:00:00       B7              Bureu    C   
441       1346  2025-03-28 06:00:00       B5  Centre commercial    D   

     temperature  humidite    co2  occupation  consommation_kwh  \
59          24.2     108.0  813.0        18.0             -15.0   
154         21.0      59.4  894.0        33.0             -50.0   
199         32.2      53.8  621.0       103.0             -20.0   
441         29.0      44.1  834.0         6.0            -100.0   

    mode_climatisation etat_systeme jour_semaine alerte  
59              Normal       Normal        Lundi    Non  
154             Normal       Normal     Dimanche    Non  
199             Normal       Normal        Jeudi    Non  
441                Eco 

##### 12) f) Si une valeur est manifestement erronée et qu’on ne peut pas retrouver sa vraie valeur, la transformer en valeur manquante 

In [19]:
nan_avant = df[["humidite", "consommation_kwh", "occupation"]].isna().sum()
print(nan_avant)

humidite            11
consommation_kwh     5
occupation           6
dtype: int64


In [20]:
df.loc[(df["humidite"] < 0) | (df["humidite"] > 100), "humidite"] = np.nan

verification

In [21]:
print(df["humidite"].isna().sum())

21


In [22]:
df.loc[(df["consommation_kwh"] < 0), "consommation_kwh"] = np.nan

verification

In [23]:
print(df["consommation_kwh"].isna().sum())

9


In [24]:
df.loc[(df["occupation"] < 0), "occupation"] = np.nan

verification

In [25]:
print(df["occupation"].isna().sum())

11


##### 12) g) rechercher des valeurs telles que catégories mal orthographiées. 

In [26]:
categorical_columns = df.select_dtypes(include=["object"]).columns.tolist()
for col in categorical_columns:
    print(f"--- {col} ---")
    print(sorted(df[col].dropna().unique()))
    print()

--- date ---
['2025-00-12', '2025-01-01 00:00:00', '2025-01-01 06:00:00', '2025-01-01 12:00:00', '2025-01-01 18:00:00', '2025-01-02 00:00:00', '2025-01-02 06:00:00', '2025-01-02 12:00:00', '2025-01-02 18:00:00', '2025-01-03 00:00:00', '2025-01-03 06:00:00', '2025-01-03 12:00:00', '2025-01-03 18:00:00', '2025-01-04 00:00:00', '2025-01-04 06:00:00', '2025-01-04 12:00:00', '2025-01-04 18:00:00', '2025-01-05 00:00:00', '2025-01-05 06:00:00', '2025-01-05 12:00:00', '2025-01-05 18:00:00', '2025-01-06 00:00:00', '2025-01-06 06:00:00', '2025-01-06 12:00:00', '2025-01-06 18:00:00', '2025-01-07 00:00:00', '2025-01-07 06:00:00', '2025-01-07 12:00:00', '2025-01-07 18:00:00', '2025-01-08 00:00:00', '2025-01-08 06:00:00', '2025-01-08 12:00:00', '2025-01-08 18:00:00', '2025-01-09 00:00:00', '2025-01-09 06:00:00', '2025-01-09 12:00:00', '2025-01-09 18:00:00', '2025-01-10 00:00:00', '2025-01-10 06:00:00', '2025-01-10 12:00:00', '2025-01-10 18:00:00', '2025-01-11 00:00:00', '2025-01-11 06:00:00', '2025-

C:\Users\dell\AppData\Local\Temp\ipykernel_6248\1380117484.py:1: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  categorical_columns = df.select_dtypes(include=["object"]).columns.tolist()


##### 12) h) normaliser les catégories textuelles en supprimant les espaces puis en uniformisant la casse 

In [29]:
colonnes_a_normaliser = ["batiment","type_batiment","zone","mode_climatisation","etat_systeme","jour_semaine","alerte"]

for col in colonnes_a_normaliser:
    df[col] = df[col].str.strip().str.lower()

In [30]:
for col in colonnes_a_normaliser:
    print(f"--- {col} ---")
    print(sorted(df[col].dropna().unique()))
    print(df[col].nunique(), "modalités")
    print()

--- batiment ---
['b1', 'b2', 'b3', 'b4', 'b5', 'b6', 'b7', 'b8']
8 modalités

--- type_batiment ---
['bureau', 'bureu', 'centre commercial', 'ecole', 'entrepot', 'entrepôt', 'hôpital', 'université', 'école']
9 modalités

--- zone ---
['a', 'b', 'c', 'd']
4 modalités

--- mode_climatisation ---
['boost', 'eco', 'normal', 'normale']
4 modalités

--- etat_systeme ---
['alerte', 'normal', 'panne']
3 modalités

--- jour_semaine ---
['dimanche', 'jeudi', 'lundi', 'mardi', 'mercredi', 'samedi', 'vendredi']
7 modalités

--- alerte ---
['non', 'oui']
2 modalités



correction des fautes de frappe 

In [33]:
corrections_type_bat = {
    "bureu": "bureau",
    "ecole": "école",
    "entrepot": "entrepôt"
}
corrections_mode_clim = {
    "normale": "normal"
}

df["type_batiment"] = df["type_batiment"].replace(corrections_type_bat)
df["mode_climatisation"] = df["mode_climatisation"].replace(corrections_mode_clim)

In [34]:
for col in colonnes_a_normaliser:
    print(f"--- {col} ---")
    print(sorted(df[col].dropna().unique()))
    print(df[col].nunique(), "modalités")
    print()

--- batiment ---
['b1', 'b2', 'b3', 'b4', 'b5', 'b6', 'b7', 'b8']
8 modalités

--- type_batiment ---
['bureau', 'centre commercial', 'entrepôt', 'hôpital', 'université', 'école']
6 modalités

--- zone ---
['a', 'b', 'c', 'd']
4 modalités

--- mode_climatisation ---
['boost', 'eco', 'normal']
3 modalités

--- etat_systeme ---
['alerte', 'normal', 'panne']
3 modalités

--- jour_semaine ---
['dimanche', 'jeudi', 'lundi', 'mardi', 'mercredi', 'samedi', 'vendredi']
7 modalités

--- alerte ---
['non', 'oui']
2 modalités



correction pour les date incoherents et invalide en les remplacant par des valeurs manquantes et de convertir les dates en type date car elles sont en type string

In [35]:
nat_avant = df["date"].isna().sum()
print("Manquants avant :", nat_avant)

Manquants avant : 0


In [36]:
df["date"] = pd.to_datetime(df["date"], format="%Y-%m-%d %H:%M:%S", errors="coerce")

In [37]:
print(df["date"].dtype)

nat_apres = df["date"].isna().sum()
print("Manquants après :", nat_apres)

datetime64[us]
Manquants après : 5


##### 13) Pour les valeurs manquantes a) Calculer le nombre et le pourcentage de valeurs manquantes par colonne 

In [41]:
nb_manquants = df.isna().sum()
pourcentage = (df.isna().mean() * 100).round(2)
resume_manquants = pd.DataFrame({
    "nb_manquants": nb_manquants,
    "pourcentage": pourcentage
}).sort_values(by="nb_manquants", ascending=False)

print(resume_manquants)

                    nb_manquants  pourcentage
humidite                      21         4.14
temperature                   12         2.37
occupation                    11         2.17
consommation_kwh               9         1.78
co2                            7         1.38
date                           5         0.99
mode_climatisation             5         0.99
jour_semaine                   5         0.99
type_batiment                  4         0.79
id_mesure                      0         0.00
zone                           0         0.00
batiment                       0         0.00
etat_systeme                   0         0.00
alerte                         0         0.00


##### 13) b) Quelle variable possède le plus de valeurs manquantes ? 

La variable humidite possede le plus de valeurs manquantes 

##### 13) c) Quelle stratégie utiliser pour les valeurs manquantes ?

La strategie a utiliser pour les valeurs manquantes est soit on les supprime s'ils ne sont pas trop influentes, oubien on les remplace par soit la moyenne, la mediane ou le mode c'est a dire la valeur la plus frequente pour les variables categorielles.

##### 13) d) Peut-on supprimer toutes les lignes contenant des valeurs manquantes ? 

Non on ne peut supprimer toutes les lignes contenant les valeurs manquantes . ca depend des conditions par exemple si on a :

--très peu de lignes concernées (moins de 5 %)

--manquants répartis au hasard

--grand jeu de données, où perdre quelques lignes ne change rien

##### 13) e) Dans quels cas utiliser la moyenne ? 

On utilise la moyenne pour remplacer des valeurs manquantes lorsque la variable est numérique et continue, que sa distribution est à peu près symétrique et qu'elle ne contient pas de valeurs aberrantes importantes, car la moyenne y est très sensible. Le pourcentage de manquants doit aussi rester faible, sinon l'imputation réduit artificiellement la variabilité des données.

##### 13) f) Quand préférer la médiane ?

on préfère la médiane quand la variable est numérique et qu'elle contient des valeurs aberrantes ou une distribution asymétrique, car elle est robuste aux extrêmes, contrairement à la moyenne.

##### 13) g) Comment traiter une variable catégorielle ?

on traite les manquants d'une variable catégorielle par le mode, par une modalité « inconnu », ou par déduction depuis d'autres colonnes, mais jamais par la moyenne ou la médiane.

##### 14) Pour les doublons a) Identifier les

In [55]:
print("le nombre de doublons est : \n", df.duplicated().sum())

le nombre de doublons est : 
 7


##### 14) Pour les doublons a) Afficher les

In [57]:
df[df.duplicated()]

,id_mesure,date,batiment,type_batiment,zone,temperature,humidite,co2,occupation,consommation_kwh,mode_climatisation,etat_systeme,jour_semaine,alerte
123,1118,2025-01-30 06:00:00,b1,bureau,a,22.0,NaN,479.0,64.0,145.0,normal,normal,jeudi,non
192,1026,2025-01-07 06:00:00,b6,bureau,a,23.0,74.3,952.0,89.0,247.6,normal,normal,mardi,oui
362,1456,2025-04-24 18:00:00,b5,centre commercial,d,26.8,61.3,843.0,59.0,186.9,normal,normal,jeudi,non
418,1402,2025-04-11 06:00:00,b2,école,a,22.6,44.0,868.0,62.0,172.5,eco,normal,vendredi,non
434,1264,2025-03-07 18:00:00,b1,bureau,b,21.1,38.3,1034.0,74.0,199.2,normal,panne,vendredi,oui
464,1320,2025-03-21 18:00:00,b7,bureau,b,27.5,27.2,359.0,33.0,185.8,normal,normal,vendredi,non
498,1479,2025-04-30 12:00:00,b3,hôpital,b,23.9,68.9,644.0,27.0,162.5,normal,normal,mercredi,non


##### 13) c) sont-ils réellement identiques ?

In [ ]:
# on Sélectionne toutes les lignes dupliquées (l'originale ET ses copies)
doublons_complets = df[df.duplicated(keep=False)]

# on Trie par les colonnes clés pour les afficher côte à côte
doublons_complets.sort_values(by=list(df.columns))


,id_mesure,date,batiment,type_batiment,zone,temperature,humidite,co2,occupation,consommation_kwh,mode_climatisation,etat_systeme,jour_semaine,alerte
117,1026,2025-01-07 06:00:00,b6,bureau,a,23.0,74.3,952.0,89.0,247.6,normal,normal,mardi,oui
192,1026,2025-01-07 06:00:00,b6,bureau,a,23.0,74.3,952.0,89.0,247.6,normal,normal,mardi,oui
9,1118,2025-01-30 06:00:00,b1,bureau,a,22.0,NaN,479.0,64.0,145.0,normal,normal,jeudi,non
123,1118,2025-01-30 06:00:00,b1,bureau,a,22.0,NaN,479.0,64.0,145.0,normal,normal,jeudi,non
23,1264,2025-03-07 18:00:00,b1,bureau,b,21.1,38.3,1034.0,74.0,199.2,normal,panne,vendredi,oui
434,1264,2025-03-07 18:00:00,b1,bureau,b,21.1,38.3,1034.0,74.0,199.2,normal,panne,vendredi,oui
455,1320,2025-03-21 18:00:00,b7,bureau,b,27.5,27.2,359.0,33.0,185.8,normal,normal,vendredi,non
464,1320,2025-03-21 18:00:00,b7,bureau,b,27.5,27.2,359.0,33.0,185.8,normal,normal,vendredi,non
238,1402,2025-04-11 06:00:00,b2,école,a,22.6,44.0,868.0,62.0,172.5,eco,normal,vendredi,non
418,1402,2025-04-11 06:00:00,b2,école,a,22.6,44.0,868.0,62.0,172.5,eco,normal,vendredi,non


Oui ils sont reellement identiques

##### 13) d) supprimer les doublons réellement identifiés. 

In [65]:
df.drop_duplicates(keep='first', inplace=True)